src/3d/qgs3daxis.cpp

```diff
@@ -101,32 +101,65 @@ bool Qgs3DAxis::eventFilter( QObject *watched, QEvent *event )
   if ( watched != mParentWindow )
     return false;
 
+  if ( event->type() == QEvent::MouseButtonPress )
+  {
+    // registrer mouse click to detect dragging
+    mHasClicked = true;
+    QMouseEvent *mouseEvent = static_cast<QMouseEvent *>( event );
+    mLastClickedPos = mouseEvent->pos();
+  }
+
   // handle QEvent::MouseButtonRelease as it represents the end of click and QEvent::MouseMove.
-  if ( event->type() == QEvent::MouseButtonRelease || event->type() == QEvent::MouseMove )
+  else if ( event->type() == QEvent::MouseButtonRelease || event->type() == QEvent::MouseMove )
   {
-    QMouseEvent *lastClickEvent = static_cast<QMouseEvent *>( event );
-    // limit ray caster usage to the axis viewport
-    QPointF normalizedPos( ( float )lastClickEvent->pos().x() / mParentWindow->width(),
-                           ( float )lastClickEvent->pos().y() / mParentWindow->height() );
-#ifdef DEBUG
-    if ( event->type() == QEvent::MouseButtonRelease )
-      qDebug() << "normalized pos:" << normalizedPos << "/ viewport:" << mAxisViewport->normalizedRect();
-#endif
-
-    if ( mAxisViewport->normalizedRect().contains( normalizedPos ) )
+    QMouseEvent *mouseEvent = static_cast<QMouseEvent *>( event );
+
+    // user has clicked and move ==> dragging start
+    if ( event->type() == QEvent::MouseMove &&
+         ( ( mHasClicked  && mouseEvent->pos() != mLastClickedPos ) || mIsDragging ) )
```

Comment:
typically in qt / qgis there is some tolerance allowed before dragging is started rather than checking exact position - it is good for UX as users sometimes inadvertently do a few pixel drag even when they meant to click. E.g. code from QgsSymbolButton:
```cpp
if ( ( e->pos() - mDragStartPosition ).manhattanLength() < QApplication::startDragDistance() )
```

Reply:
great! I was looking for something like that!